common/app/model/content/Atoms.scala

```diff
@@ -241,6 +236,19 @@ object MediaAtom extends common.Logging {
     )
   }
 
+  def activeYouTubeAsset(atom: MediaAtom): Option[MediaAsset] = {
+    val defaultAsset = atom.assets.headOption
+
+    val activeAsset = for {
+      version <- atom.activeVersion
+      asset <- atom.assets.find(_.version == version)
+    } yield asset
+
+    val asset = activeAsset orElse defaultAsset
+
+    // sanity check the platform to avoid template rendering errors
+    asset.filter(_.platform == MediaAssetPlatform.Youtube)
+  }
```

Comment:
I guess you can chain everything to avoid assigning intermediary val
```
(for {
      version <- atom.activeVersion
      asset <- atom.assets.find(_.version == version)
    } yield asset)
    .orElse(atom.assets.headOption)
    .filter(_.platform == MediaAssetPlatform.Youtube) // sanity check the platform to avoid template rendering errors
```